### PR TITLE
feat: integrate with web2 auth API

### PR DESCRIPTION
**Pull request by quanta-kt**
## Release Notes

- **New Features**
  - Integrated query management capabilities into the application.
  - Enhanced login and signup functionalities with improved user experience, including local state management and loading indicators.
  - Introduced loading state visualization for buttons during operations.
  - Added a new environment variable for API endpoint configuration.

- **Bug Fixes**
  - Improved error handling for login and signup processes.

- **Chores**
  - Added new dependencies: `@tanstack/react-query` and `axios` for enhanced functionality.

### CODE REVIEW

File: net/client.ts
Comment (rdj68) [inline net/client.ts:1]: We should use environment vars for base url and network

File: app/auth/web2auth.tsx [hunk 1]
```diff
@@ -104,12 +149,53 @@ function Login() {
...
+        console.log(e.response?.data);
+      }
+
+      // FIXME: report errors to UI
+
+      return;
+    }
+
+    router.dismissAll();
+    router.replace("/");
```
Comment (rdj68) [inline app/auth/web2auth.tsx:179]: Shouldn't you redirect for succesfull signup only?? I see that we check for need of auth in home layout, it's fine if you have scheduled these changes for a later PR.

> // FIXME: *actually* perform a check if user is logged in

> const needsLogin = true;"

File: net/queries/useWeb2Signup.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import client from "../client";
+import { MutationBuilder } from "./builder";
+
+export interface Web2SignupRequest {
+  email: string;
+  username: string;
+  password: string;
+}
+
+const useWeb2Login = new MutationBuilder<Web2SignupRequest>(client)
```
Comment (rdj68) [inline net/queries/useWeb2Signup.ts:10]: Rename this to useWeb2SignUp as suggested by the bot.

File: app/auth/web2auth.tsx (hunk 1 above)
Comment (quanta-kt) [reply app/auth/web2auth.tsx:179]: > Shouldn't you redirect for succesfull signup only??
That is what the `return` statement in `catch` block is for

> I see that we check for need of auth in home layout
That will require knowing if the user is signed-in, which in turn requires a persistent storage to store current auth details.

> it's fine if you have scheduled these changes for a later PR.
Yeah. With the SQLite setup.

File: net/queries/useWeb2Signup.ts (hunk 2 above)
Comment (quanta-kt) [reply net/queries/useWeb2Signup.ts:10]: oops, will fixup.

**Pull request merged**